react-context/flow.txt

```diff
@@ -0,0 +1,22 @@
+User can sign up.
+  - The input fields page is rendered and depending on what the action prop is, the href and text content for appropriate rendered elements.
+  - When the user enters a username and password into the input field, the data is stored in the state of the AuthForm component
+  - Submitting the form triggers the client sends a request to the server to input a new user into the database
+  - A condition is checked whether the username exists in the database or not
+  - if it does not, it inserts it into the database with a query using SQL
+  - the username is added with a hashedPassword created as well
+  - the user receives a token that is unique to that account and is needed to sign in
+User can sign in.
+  - Once the user has succesfully signed up, signing in requires an authentication
+  - The authentication is checked via fetch request and returns a match for the username and password
+  - Signing in sends the token back to the server so that the server can verify it
+  - the token is stored in the localStorage of the app and the state of the App class is updated to contain the information about the user
+  - if the token returns null when decoded, the authorizing property in the state is set to false.
```

Comment:
This is actually part of the "User can stay signed in" feature. The app initially starts as `isAuthorizing: true` because the app has not found or decoded the token yet.

Component lifecycle goes like this:

```
constructor -> render -> componentDidMount -> ... -> setState -> render
```

`componentDidMount` in App only runs _one_ time for the whole lifetime of the app right after the page loads. Once the App has decided a) there's no token or b) the token was found and decoded, then it is "finished" authorizing, whether there's a user or not.